projects/packages/phan-plugins/src/HtmlTemplatePlugin.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * Phan plugin for handling "html template" files.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\PhanPlugins;
+
+use ast\Node;
+use Phan\CodeBase;
+use Phan\Language\Context;
+use Phan\Language\Element\Variable;
+use Phan\Language\FQSEN\FullyQualifiedFunctionName;
+use Phan\Language\FQSEN\FullyQualifiedMethodName;
+use Phan\Language\Scope\ClassScope;
+use Phan\Language\Scope\FunctionLikeScope;
+use Phan\Language\Type;
+use Phan\Language\Type\StaticType;
+use Phan\Language\UnionType;
+use Phan\PluginV3;
+use Phan\PluginV3\BeforeAnalyzeFileCapability;
+use PhpToken;
+
+/**
+ * Phan plugin for handling "html template" files.
+ */
+class HtmlTemplatePlugin extends PluginV3 implements BeforeAnalyzeFileCapability {
+
+	private const IDENTIFIER  = '[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*';
```

Comment:
Tiny nitpick: I suspect we don't want to include `\x7f` (`DEL`). PHP documentation seems to confirm that:
https://www.php.net/manual/en/language.variables.basics.php

```suggestion
	private const IDENTIFIER  = '[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*';
```

Reply:
I copied the regex from Phan's own code. Turns out it was allowed [until PHP 7.1](https://www.php.net/manual/en/migration71.incompatible.php#migration71.incompatible.delete-control-character-in-identifiers).

Reply:
So you mean I spent all that time reviewing the regex for naught? 😆

Reply:
You might have caught a bug in Phan by reviewing it. 😀